### PR TITLE
Cleanup csv generator

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
Cleanup code in `csv-generator`.

**Release note**:
```release-note
None
```

### CODE REVIEW

File: hack/csv-generator.go [hunk 1]
```diff
@@ -104,129 +104,89 @@ func init() {
...
+			return err
+		}
+	}
+
+	return nil
+}
+
+func generateCsv() error {
+	csv := &csvv1.ClusterServiceVersion{}
+	err := readFileToObject(f.file, csv)
```
Comment (0xFelix) [inline hack/csv-generator.go:122]: Inline this check?

File: hack/csv-generator.go [hunk 2]
```diff
@@ -104,129 +104,89 @@ func init() {
...
 	if err != nil {
 		return err
 	}
 
-	err = replaceVariables(f, &csv)
+	err = replaceVariables(f, csv)
```
Comment (0xFelix) [inline hack/csv-generator.go:127]: Same

File: hack/csv-generator.go [hunk 3]
```diff
@@ -104,129 +104,89 @@ func init() {
...
 
-		err = readAndDecodeToCRD(fsInfo, &crd)
+		err = readFileToObject(fsInfo.Name(), crd)
 		if err != nil {
 			return err
 		}
 
-		err = marshallObject(crd, nil, os.Stdout)
+		cleanupCrd(crd)
+		err = writeObjectYaml(crd, os.Stdout)
```
Comment (0xFelix) [inline hack/csv-generator.go:162]: Same

**Review by 0xFelix — COMMENTED** [review]
Inline check in L62 too?

File: hack/csv-generator.go (hunk 3 above)
Comment (akrejcir) [reply hack/csv-generator.go]: done

File: hack/csv-generator.go (hunk 2 above)
Comment (akrejcir) [reply hack/csv-generator.go]: done

File: hack/csv-generator.go (hunk 1 above)
Comment (akrejcir) [reply hack/csv-generator.go]: done

**Pull request closed without merging**